File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/profile/ValidateCodeForChangeMailUseCase.java
package com.lattis.ellipse.domain.interactor.profile;

import com.lattis.ellipse.data.UserDataRepository;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.UserRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

public class ValidateCodeForChangeMailUseCase extends UseCase<Boolean> {

    private UserRepository userRepository;

    private String code;
    private String email;

    public ValidateCodeForChangeMailUseCase withCode(String code) {
        this.code = code;
        return this;
    }
    public ValidateCodeForChangeMailUseCase withEmail(String email) {
        this.email = email;
        return this;
    }

    @Inject
    public ValidateCodeForChangeMailUseCase(ThreadExecutor threadExecutor,
                                            PostExecutionThread postExecutionThread,
                                            UserDataRepository userDataRepository
                                            ) {
        super(threadExecutor, postExecutionThread);
        this.userRepository = userDataRepository;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return userRepository.validateCodeForChangeEmail(code,email);
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/user/ValidateCodeForChangeMailUseCase.kt
package com.lattis.domain.usecase.user

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.UserRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class ValidateCodeForChangeMailUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val userRepository: UserRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var code: String? = null
    private var email: String? = null
    fun withCode(code: String?): ValidateCodeForChangeMailUseCase {
        this.code = code
        return this
    }

    fun withEmail(email: String?): ValidateCodeForChangeMailUseCase {
        this.email = email
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return userRepository.validateCodeForEmailChange(code!!, email!!)
    }

}